pkg/controller/nodeipam/ipam/cidr_allocator.go

```diff
@@ -128,11 +130,15 @@ func New(kubeClient clientset.Interface, cloud cloudprovider.Interface, nodeInfo
 	case RangeAllocatorType:
 		return NewCIDRRangeAllocator(kubeClient, nodeInformer, allocatorParams, nodeList)
 	case MultiCIDRRangeAllocatorType:
-		cccList, err := listClusterCIDRs(kubeClient)
+		if !utilfeature.DefaultFeatureGate.Enabled(features.MultiCIDRRangeAllocator) {
```

Comment:
You're going to hate me, but this needs to be back-integrated with the previous commits (or we accept that this PR needs a full squash).  I don't want ungated code to be committed, in case of a rollback or something

Reply:
No worries, Done